docs/controller.md

```diff
@@ -0,0 +1,108 @@
+# Architecture Document for Controller Component
+
+## Description
+The Controller is the component giving meaning to the [pipeline](#structure) file given by the Monitor. It receives [pipelines](#pipeline), parse them into a set of [actions](#actions) and send these actions sequentially to the Scheduler, for each of these actions, the Scheduler **must** notify the Controller when a job has been scheduled and has been completed successfully or encountered an error, it can be done just by sending the logs from the Agent for instance. Thanks to these information, the Scheduler is able to provide information about a pipeline state to anyone (the Monitor or any other client).
+## Features
+- Users can send pipelines containing actions to execute. These actions are basically shell commands to execute. These pipelines are `yaml` files.
```

Comment:
Actions could be shell commands or something else in the future. Let's not get into too many details for now.

```suggestion
- Users send pipelines containing actions to execute. Pipelines are described through YAML formatted files (TODO define those YAML). 
```

Reply:
We propose a reference for these manifest in the `structure` section. To make this paragraph clearer, we could add a link to the `structure` section or move the `structure` paragraph to the beginning of this doc

Reply:
ack

Reply:
Done, we renamed the structure section into Pipeline YAML reference